File: C#/73dotnet/73dotnet/The73CalculatorTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;

namespace _73dotnet
{
    [TestClass]
    public class The73CalculatorTest
    {
        static new readonly Dictionary<string, Func<int, int>> testMethods = new Dictionary<string, Func<int, int>>()
        {
            { "UseIf", The73Calculator.useif},
            { "useifeq", The73Calculator.useifeq},
            { "useXor", The73Calculator.useXor},
            { "useArray", The73Calculator.useArray},
            { "usingSwitch", The73Calculator.usingSwitch},
            { "useSubstraction", The73Calculator.useSubstraction},
            { "useDivision", The73Calculator.useDivision},
            { "useStringReplaceParseInt", The73Calculator.useStringReplaceParseInt},
            { "useModulo", The73Calculator.useModulo},
            { "useObjectMap", The73Calculator.useHashMap},
            { "useIndexerDictionary", The73Calculator.useIndexerDictionary},
            { "usePolynom", The73Calculator.usePolynom}

        };

        [DataRow("UseIf")]
        [DataRow("useifeq")]
        [DataRow("useXor")]
        [DataRow("useArray")]
        [DataRow("usingSwitch")]
        [DataRow("useSubstraction")]
        [DataRow("useDivision")]
        [DataRow("useStringReplaceParseInt")]
        [DataRow("useModulo")]
        [DataRow("useObjectMap")]
        [DataRow("useIndexerDictionary")]
        [DataRow("usePolynom")]
        [DataTestMethod]

        public void TestThe73Methods(string methodNameToeTest)
        {
            var method = testMethods[methodNameToeTest];
            TestTheMethod(method);
        }

        public void TestTheMethod(Func<int, int> f)
        {
            for (int i = 0; i < 100000; i++)
            {
                Assert.AreEqual(7, f(3));
                Assert.AreEqual(3, f(7));
            }
        }
    }
}

File: javascript/The73Calculator.test.js
const The73Calculator = require("./The73Calculator.js");


const testMEthods = [
    The73Calculator.useif,
    The73Calculator.useifeq,
    The73Calculator.useXor,
    The73Calculator.useArray,
    The73Calculator.useSwitch,
    The73Calculator.useSubstraction,
    The73Calculator.useDivision,
    The73Calculator.useStringReplaceParseInt,
    The73Calculator.useStringReplaceNumber,
    The73Calculator.useModulo,
    The73Calculator.useObjectMap,
    The73Calculator.useIndexerDictionary,
    The73Calculator.usePolynom,
    The73Calculator.useOctalBase
];

describe.each(testMEthods)('With params for %s', (methodUnderTest) => {
    it(`${methodUnderTest.name} test`, () => {
        expectBehavior(methodUnderTest);
    });
});

function expectBehavior(f) {
    expect(f(3)).toBe(7);
    expect(f(7)).toBe(3);
}